### ISSUE TITLE
Basic Authentication

### ISSUE THREAD

**Issue by EricAndrechek** [kofigumbs/multi#47]
Hi there!
I am using multi for some web-apps I have written myself. My very basic web app is protected by basic authentication on Nginx. In Chrome or Safari when opening my webpage for the first time, I get a popup asking me for a username and password to access this site. If the login credentials are incorrect it returns an Unauthorized error via Nginx, but if correct I can access my content.

When launching a multi app protected by basic authentication, there is no prompt for log-in details and it instead immediately shows an unauthorized error. My current solution to the problem was to open the web app in Safari and login and save the login to my keychain which then allowed me to access the site in multi.

**Comment by kofigumbs** [comment]
That's a cool use of Multi! Glad you found a workaround for now.

_Note to my future self: [WebKit auth documentation](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455638-webview)_

**Comment by mantasio** [comment]
@EricAndrechek You could try configuring your Multi app URL to `http(s)://username:password@yourwebsite.com` format in order to skip the login prompt, but I guess you don't want that. Not entirely sure if it would work either.

**Comment by EricAndrechek** [comment]
I tried that and it didn’t seem to work, I suspect it is because the actual webpage I am loading is not password protected, but rather a specific resource it is is password protected.

**Comment by kofigumbs** [comment]
I released support for this in [2.1.0](https://github.com/kofigumbs/multi/releases/tag/v2.1.0). Here's an example config that uses basic auth:

```json
{
  "tabs": [{
    "title": "Basic Auth",
    "basicAuthUser": "guest",
    "basicAuthPassword": "guest",
    "url": "https://jigsaw.w3.org/HTTP/Basic/"
  }]
}
```
